app/Jobs/GenerateHashValue.php

```diff
@@ -0,0 +1,226 @@
+
+    }
+
+    /**
+     * Gets a thumbnail for a single image.
+     *
+     * @param Image $image
+     * @return string byte string of the image
+     */
+    protected function getThumbnail(Image $image)
+    {
+        $prefix = fragment_uuid_path($image->uuid);
+        $format = config('thumbnails.format');
+
+        return Storage::disk(config('thumbnails.storage_disk'))
+            ->get("{$prefix}.{$format}");
+    }
+
+    /**
+     * Execute the HashValueGenerator and get the resulting hash value to the image.
+     *
+     * @param Image $image
+     * @return string
+     * @throws Exception
+     */
+    protected function getHashValue(Image $image)
+    {
+        $imageByteString = $this->getThumbnail($image);
+        dd($imageByteString);
+
+        $script = config('biigle.hash_value_generator');
+
+        try {
+            $inputPath = $this->createInputJson($image, $imageByteString);
+            $outputPath = $this->getOutputJsonPath($image);
+            $output = $this->python("{$script} {$inputPath} {$outputPath}");
+            $hashValue = json_decode(File::get($outputPath), true);
+        } catch (Exception $e) {
+            $input = File::get($inputPath);
+            throw new Exception("Input: {$input}\n" . $e->getMessage());
+        } finally {
+            if (isset($inputPath)) {
+                $this->maybeDeleteFile($inputPath);
+            }
+
+            if (isset($outputPath)) {
+                $this->maybeDeleteFile($outputPath);
+            }
+        }
+        return $hashValue;
+
+
+    }
+
+
+
+    /**
+     * Execute a Python script.
+     *
+     * @param $command
+     * @return string the hash value for the image
+     * @throws Exception
+     */
+    protected function python($command)
+    {
+        $lines = 0;
+        $code = 0;
+        $python = config('biigle.python');
+
+
+        exec("{$python} {$command}", $lines, $code);
+
+        if ($code !== 0) {
+            throw new Exception("Error while executing Python script':\n".implode("\n", $lines));
+        }
+
+        return end($lines);
+
+    }
+
+    /**
+     * Get the path to to input file for the object tracking script.
+     *
+     * @param Image $image
+     *
+     * @return string
+     */
+    protected function getInputJsonPath(Image $image)
+    {
+        return config('hash.tmp_dir')."/generate_hash_value_input_{$image->id}.json";
+    }
+
+    /**
+    * Create the JSON file that is the input for the HashValueGenerator script.
+    *
+    * @param Image $image
+    * @param string $imagePath Path to the video file.
+    *
+    * @return string Path to the JSON file.
+    */
+    protected function createInputJson(Image $image, $imageAsByteString)
+    {
+        $path = $this->getInputJsonPath($image);
+        $content = json_encode([
+            'image_as_byte_string' => $imageAsByteString,
+            'image_id' => $image->id,
+        ]);
+
+        File::put($path, $content);
```

Comment:
While Testing I got the problem that this Line doesn't work do you understand why?

Reply:
do you understand why this does not work @mzur ?

Reply:
Whe I run the tests, I get this error:
```
1) Biigle\Tests\Jobs\GenerateHashValueTest::testHandle
Illuminate\Contracts\Filesystem\FileNotFoundException: File not found at path: b3/3e/b33ece29-8b5a-31ce-b167-76e00978cad8.jpg

/var/www/vendor/laravel/framework/src/Illuminate/Filesystem/FilesystemAdapter.php:135
/var/www/app/Jobs/GenerateHashValue.php:104
/var/www/app/Jobs/GenerateHashValue.php:116
/var/www/app/Jobs/GenerateHashValue.php:83
/var/www/tests/php/Jobs/GenerateHashValueTest.php:27
```

That's because the thumbnail file does not exist in the test. You should:

1. Handle the actual case where the thumbnail does not exist (and skip generating the hash).
2. Create a mock thumbnail file in the test (e.g. with a string content `"abcd"`).